src/front-door/azext_front_door/custom.py

```diff
@@ -519,14 +515,30 @@ def create_fd_routing_rules(cmd, resource_group_name, front_door_name, item_name
                       '[--custom-fragment CUSTOM_FRAGMENT] [--custom-query-string CUSTOM_QUERY_STRING]')
 
     # pylint: disable=line-too-long
-    if (route_type == 'Forward' and any([custom_host, custom_path, custom_fragment, custom_query_string]) and getattr(redirect_type, 'is_default', None) and getattr(redirect_protocol, 'is_default', None)):
+    if 'Forward' in route_type and any([custom_host, custom_path, custom_fragment, custom_query_string]) and getattr(redirect_type, 'is_default', None) and getattr(redirect_protocol, 'is_default', None):
         from knack.util import CLIError
         raise CLIError(forwarding_usage)
-    if route_type == 'Redirect' and any([custom_forwarding_path, forwarding_protocol, backend_pool,
+    if 'Redirect' in route_type and any([custom_forwarding_path, forwarding_protocol, backend_pool,
                                          query_parameter_strip_directive, dynamic_compression]):
         from knack.util import CLIError
         raise CLIError(redirect_usage)
 
+
+def create_fd_routing_rules(cmd, resource_group_name, front_door_name, item_name, frontend_endpoints, route_type,
```

Comment:
shoud create method support rules_engine too?

Reply:
Yeah, I guess you are right. Rules engine has to be created first and then linked to a route.  'create' word in function name threw me off, hence I added it only for the update routing rule method. In this case we can actually create a new route in a front door that can link an existing rules engine, I'll add it.

Reply:
done in next commit